R/utils-schema-versions.R

```diff
@@ -0,0 +1,203 @@
+#' Get hub config schema versions
+#'
+#' @param config A `<config>` class object. Usually the output of `read_config` or `read_config_file`.
+#' @inherit extract_schema_version return
+#'
+#' @export
+#' @describeIn get_version_config Get schema version from config list representation.
+#' @examples
+#' config <- read_config_file(
+#'   system.file("config", "tasks.json", package = "hubUtils")
+#' )
+#' get_version_config(config)
+get_version_config <- function(config) {
+  schema_version <- config$schema_version
+  if (is.null(schema_version)) {
+    cli::cli_abort(c("x" = "No {.field schema_version} property found in {.arg config}."))
+  }
+  extract_schema_version(config$schema_version)
+}
+
+#' @inheritParams read_config_file
+#' @export
+#' @describeIn get_version_config Get schema version from config file at specific path.
+#'
+#' @examples
+#' config_path <- system.file("config", "tasks.json", package = "hubUtils")
+#' get_version_file(config_path)
+get_version_file <- function(config_path) {
+  checkmate::assert_file_exists(config_path)
+  read_config_file(config_path) |>
+    get_version_config()
+}
+
+#' @inheritParams read_config
+#' @param config_type Character vector specifying the type of config file to read.
+#' One of "tasks" or "admin". Default is "tasks".
+#' @export
+#' @describeIn get_version_config Get schema version from config file at specific path.
+#'
+#' @examples
+#' hub_path <- system.file("testhubs/simple", package = "hubUtils")
+#' get_version_hub(hub_path)
+#' get_version_hub(hub_path, "admin")
+get_version_hub <- function(hub_path, config_type = c("tasks", "admin")) {
+  config_type <- rlang::arg_match(config_type)
+  read_config(hub_path, config_type) |>
+    get_version_config()
+}
+
+## COMPARE VERSION NUMBERS -----------------------------------------------------
+#' Compare hub config `schema_version`s to specific version numbers from
+#' a variety of sources
+#'
+#' @param config A `<config>` class object. Usually the output of `read_config` or `read_config_file`.
+#' @param version Character string. Version number to compare against.
+#' @param schema_version Character string. A config `schema_version` property to
+#' compare against.
+#' @return `TRUE` or `FALSE` depending on how the schema version compares to the
+#' version number specified.
+#'
+#' @export
+#' @inheritParams read_config_file
+#' @inheritParams read_config
+#' @export
+#' @describeIn version_equal Check whether a schema version property is equal
+#' to a specific version number.
+#'
+#' @examples
+#' hub_path <- system.file("testhubs/simple", package = "hubUtils")
+#' config_path <- system.file("config", "tasks.json", package = "hubUtils")
+#' config <- read_config_file(config_path)
+#' schema_version <- config$schema_version
+#' # Check whether schema_version equal to v3.0.0
+#' version_equal("v3.0.0", config = config)
+#' version_equal("v3.0.0", config_path = config_path)
+#' version_equal("v3.0.0", hub_path = hub_path)
```

Comment:
I wonder if it make sense to say if the tests are expected to fail or pass for each case to avoid confusion? Because some passes or some fails here, no?